### PR TITLE
Fixed issue #8376

### PR DESCRIPTION
**Pull request by kennycastro007**
I thought about making the error reporting a function to avoid rewriting it 6 times but decided against it since I wasn't sure if it was a good idea and would've only been used in this one file

### CODE REVIEW

**Comment by ZehMatt** [comment]
Nothing speaks against a static function for when you know you will have duplicate code.

**Comment by kennycastro007** [comment]
I just changed the code to use a static function `show_error(uint8_t, const rct_string_id&)` to use instead, this would be better should the manner in which errors are handled here can be changed easily

File: src/openrct2/rct2/S6Importer.cpp [hunk 1]
```diff
@@ -1528,6 +1528,20 @@ std::unique_ptr<IParkImporter> ParkImporter::CreateS6(IObjectRepository& objectR
     return std::make_unique<S6Importer>(objectRepository);
 }
 
+static void show_error(uint8_t gErrorType, const rct_string_id& gErrorStringId)
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: We only prefix globals with g. We normally use camelCase for variable names.

File: src/openrct2/rct2/S6Importer.cpp [hunk 2]
```diff
@@ -1528,6 +1528,20 @@ std::unique_ptr<IParkImporter> ParkImporter::CreateS6(IObjectRepository& objectR
...
+{
+    rct_string_id title_text = STR_UNABLE_TO_LOAD_FILE;
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: camelCase name

File: src/openrct2/rct2/S6Importer.cpp [hunk 3]
```diff
@@ -1546,19 +1560,24 @@ void load_from_sv6(const char* path)
     }
     catch (const ObjectLoadException&)
     {
-        gErrorType = ERROR_TYPE_FILE_LOAD;
-        gErrorStringId = STR_FILE_CONTAINS_INVALID_DATA;
+        uint8_t gErrorType = ERROR_TYPE_FILE_LOAD;
+        rct_string_id gErrorStringId = STR_FILE_CONTAINS_INVALID_DATA;
+        show_error(gErrorType, gErrorStringId);
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: Why not pass the directly the value to the parameter? You don't have to assign variables first.

File: src/openrct2/rct2/S6Importer.cpp [hunk 4]
```diff
@@ -1546,19 +1560,24 @@ void load_from_sv6(const char* path)
...
     }
     catch (const IOException& loadError)
     {
-        gErrorType = ERROR_TYPE_FILE_LOAD;
-        gErrorStringId = STR_GAME_SAVE_FAILED;
+        uint8_t gErrorType = ERROR_TYPE_FILE_LOAD;
+        rct_string_id gErrorStringId = STR_GAME_SAVE_FAILED;
         log_error("Error loading: %s", loadError.what());
+
+        show_error(gErrorType, gErrorStringId);
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: Same as above

File: src/openrct2/rct2/S6Importer.cpp [hunk 5]
```diff
@@ -1546,19 +1560,24 @@ void load_from_sv6(const char* path)
...
     }
     catch (const std::exception&)
     {
-        gErrorType = ERROR_TYPE_FILE_LOAD;
-        gErrorStringId = STR_FILE_CONTAINS_INVALID_DATA;
+        uint8_t gErrorType = ERROR_TYPE_FILE_LOAD;
+        rct_string_id gErrorStringId = STR_FILE_CONTAINS_INVALID_DATA;
+
+        show_error(gErrorType, gErrorStringId);
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: Same as above

File: src/openrct2/rct2/S6Importer.cpp [hunk 6]
```diff
@@ -1584,20 +1603,26 @@ void load_from_sc6(const char* path)
     }
     catch (const ObjectLoadException& loadError)
     {
-        gErrorType = ERROR_TYPE_FILE_LOAD;
-        gErrorStringId = STR_GAME_SAVE_FAILED;
+        uint8_t gErrorType = ERROR_TYPE_FILE_LOAD;
+        rct_string_id gErrorStringId = STR_GAME_SAVE_FAILED;
         log_error("Error loading: %s", loadError.what());
+
+        show_error(gErrorType, gErrorStringId);
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: Same as above

File: src/openrct2/rct2/S6Importer.cpp [hunk 7]
```diff
@@ -1584,20 +1603,26 @@ void load_from_sc6(const char* path)
...
     }
     catch (const IOException& loadError)
     {
-        gErrorType = ERROR_TYPE_FILE_LOAD;
-        gErrorStringId = STR_GAME_SAVE_FAILED;
+        uint8_t gErrorType = ERROR_TYPE_FILE_LOAD;
+        rct_string_id gErrorStringId = STR_GAME_SAVE_FAILED;
         log_error("Error loading: %s", loadError.what());
+
+        show_error(gErrorType, gErrorStringId);
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: Same as above

File: src/openrct2/rct2/S6Importer.cpp [hunk 8]
```diff
@@ -1584,20 +1603,26 @@ void load_from_sc6(const char* path)
...
     }
     catch (const std::exception&)
     {
-        gErrorType = ERROR_TYPE_FILE_LOAD;
-        gErrorStringId = STR_FILE_CONTAINS_INVALID_DATA;
+        uint8_t gErrorType = ERROR_TYPE_FILE_LOAD;
+        rct_string_id gErrorStringId = STR_FILE_CONTAINS_INVALID_DATA;
+
+        show_error(gErrorType, gErrorStringId);
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: Same as above

File: src/openrct2/rct2/S6Importer.cpp (hunk 3 above)
Comment (kennycastro007) [reply src/openrct2/rct2/S6Importer.cpp]: Fixed it in a new commit

**Comment by kennycastro007** [comment]
@ZehMatt Is that all the changes you wanted me to make?

File: src/openrct2/rct2/S6Importer.cpp [hunk 9]
```diff
@@ -1528,6 +1528,20 @@ std::unique_ptr<IParkImporter> ParkImporter::CreateS6(IObjectRepository& objectR
     return std::make_unique<S6Importer>(objectRepository);
 }
 
+static void show_error(uint8_t errorType, const rct_string_id& errorStringId)
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: Why did you choose to use ```const rct_string_id&``` here? You should pass small things like integers always by value, this is not performance critical code but its generally better to do.

File: src/openrct2/rct2/S6Importer.cpp [hunk 10]
```diff
@@ -1528,6 +1528,20 @@ std::unique_ptr<IParkImporter> ParkImporter::CreateS6(IObjectRepository& objectR
...
+{
+    rct_string_id titleText = STR_UNABLE_TO_LOAD_FILE;
+    rct_string_id bodyText = errorStringId;
+    if (errorType == ERROR_TYPE_GENERIC)
+    {
+        titleText = errorStringId;
+        bodyText = 0xFFFF;
+    }
+    errorType = ERROR_TYPE_NONE;
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]: This assignment seems redundant.

Comment (kennycastro007) [reply src/openrct2/rct2/S6Importer.cpp]: It was left over from when I copy-pasted the common parts out, I've taken it out in my latest commit

File: src/openrct2/rct2/S6Importer.cpp (hunk 9 above)
Comment (kennycastro007) [reply src/openrct2/rct2/S6Importer.cpp]: I can see it being simpler to read and since the copy for ints is negligible, but is there another reason why it's better to pass by value? Also do you think the function should be inline?

**Comment by ZehMatt** [comment]
Given that its your first time contributing you should add yourself to the contributors in 
 'contributors.md'

**Comment by kennycastro007** [comment]
I went ahead and added myself to contributors.md, thanks for the reminder!

File: .vscode/c_cpp_properties.json [hunk 11]
```diff
@@ -49,7 +49,8 @@
             },
             "compilerPath": "/usr/bin/clang",
             "cStandard": "c11",
-            "cppStandard": "c++17"
+            "cppStandard": "c++17",
+            "configurationProvider": "vector-of-bool.cmake-tools"
```
Comment (ZehMatt) [inline .vscode/c_cpp_properties.json]: Please revert this change, its unrelated.

**Comment by kennycastro007** [comment]
Sorry about that, that was CMake Tools, I accidentally staged that change and pushed it in
